server/events/apply_command_runner.go

```diff
@@ -2,14 +2,49 @@ package events
 
 import (
 	"github.com/runatlantis/atlantis/server/events/db"
+	"github.com/runatlantis/atlantis/server/events/locking"
 	"github.com/runatlantis/atlantis/server/events/models"
 	"github.com/runatlantis/atlantis/server/events/vcs"
 )
 
+//go:generate pegomock generate -m --use-experimental-model-gen --package mocks -o mocks/mock_apply_command_locker.go ApplyCommandLocker
+
+type ApplyCommandLocker interface {
+	IsDisabled(ctx *CommandContext) bool
```

Comment:
ie. This struct isn't responsible for performing the locking, it's responsible for checking whether a lock exists. Confusing naming imo.

Id suggest renaming this to `GlobalApplyLockChecker` with the method `isLocked` (disabled also doesn't fit well if the name of the struct doesn't include that)